### ISSUE TITLE
单机训练报错Check failed: numVecSlots_ == sample.vector_slots_size() (1 vs. 0)

### ISSUE THREAD

**Issue by XinyZh** [PaddlePaddle/Paddle#2435]
使用paddle v1
样本只有一个slot，文件格式：
0;0 3.0 7.0 1.0 10.0 11.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.3 0.0 0.0 0.0 0.0 0.0 0.7 0.0 0.0 0.0 0.0 0.0 0.9091 0.0 0.0 0.0 0.0 0.0 0.0909 0.0 0.0 0.0 0.0 0.0 9.0 0.0 0.0 0.0 0.0 0.0 -0.5714 0.0 0.0 0.0 0.0 0.0 10.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 2009.0 1.0 0.0 100.0 -1.0 -1.0 2009.0 -1.0 1.0 -1.0 2615.0 2431.0 3.0 1.0 147.0 1.0 0.0 0.0 0.0 1.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 52.0 75.0 62.0 189.0 0.0 0.0 0.724867724868 0.149465046474 0.0 0.0 0.0 0.0 0.0 0.0;
通过preprocess处理成pb格式，并经validate验证，显示“ALL PASS”
train/local目录下，cpu_pb.json配置如下：
{
    "common" : {
        "cpu_gpu" : "cpu",
        "local" : true,
        "evaluation" : false,
        "data_path" : {
            "train_data_path" : [ "xxx"],
            "test_data_path" : [ "xxx"],
            "output_path" : "./result"
        },
        "train_arg" : "--saving_period=5 --port=7164 --ports_num=1 --parallel_thread_num=10 --local=0 --comment=$comment --dot_period=
100000 --log_period=10000 --num_passes=10000 --trainer_count=4  --local=false",
        "test_arg" : "--job=test --test_pass=0 --test_wait=1 --dot_period=1 --log_period=1000 --saving_period=5 --local=1 --num_passes
=1000 --start_pserver=0"
    },
    "algorithm" :
    {
        "name" : "sgd",
        "parameters" :
        {
             "num_batches_per_send_parameter" : 2,
             "num_batches_per_get_parameter" : 1,
             "batch_size" : 100,
             "learning_rate" : 0.00005,
             "learning_rate_decay_a" : 1e-5,
             "learning_rate_decay_b" : 0.0
        }
    },
    "data" :
    {
        "type" : "proto"
    },

   "network" :
    {
         "inputs" : ["input1", "label"],
         "outputs" : ["cost"],
         "layers" :
         [
              {
                   "name" : "input1",
                   "type" : "data",
                   "size" : 142
              },
              {
                   "name" : "layer1",
                   "type" : "fc",
                   "size" : 20,
                   "active_type" : "tanh",
                   "bias" : {
                                 "parameter_name" : "_layer1.bias"
                            },
                   "inputs" :
                   [
                        {
                             "name" : "input1",
                             "parameter_name" : "_layer1.w"
                        }
                   ]
              },
              {
                   "name" : "output",
                   "type" : "fc",
                   "size" : 2,
                   "active_type" : "softmax",
                   "bias" : {
                                "parameter_name" : "_output.bias"
                            },
                   "inputs" :
                   [
                       {
                           "name" : "layer1",
                           "parameter_name" : "_output.w"
                       }
                   ]
              },
              {
                  "name" : "label",
                  "type" : "data",
                  "size" : 1
              },
              {
                  "name" : "cost",
                  "type" : "multi-class-cross-entropy",
                  "inputs" :
                  [
                    {
                        "name" : "output"
                    },
                    {
                        "name" : "label"
                    }
                  ]
              },
              {
                  "name" : "auc",
                  "type" : "auc-validation",
                  "inputs" :
                  [
                      {
                          "name" : "output"
                      },
                      {
                          "name" : "label"
                      }
                  ]
              }
         ]
    }
}

运行 sh local_train.sh run cpu_pb.json，输出如下错误：
[WARNING 2017-06-10 12:50:43,415 trainer_config.conf:6] Deprecated: use Settings(learning_rate_decay_a=1e-05, ...) instead
[WARNING 2017-06-10 12:50:43,415 trainer_config.conf:7] Deprecated: use Settings(learning_rate_decay_b=0.0, ...) instead
[WARNING 2017-06-10 12:50:43,415 trainer_config.conf:8] Deprecated: use Settings(learning_rate=5e-05, ...) instead
[WARNING 2017-06-10 12:50:43,415 trainer_config.conf:9] Deprecated: use Settings(batch_size=100, ...) instead
[WARNING 2017-06-10 12:50:43,416 trainer_config.conf:10] Deprecated: use Settings(algorithm=sgd, ...) instead
[WARNING 2017-06-10 12:50:43,416 trainer_config.conf:11] Deprecated: use Settings(num_batches_per_send_parameter=2, ...) instead
[WARNING 2017-06-10 12:50:43,416 trainer_config.conf:12] Deprecated: use Settings(num_batches_per_get_parameter=1, ...) instead
F0610 12:50:46.146634 13370 ProtoDataProvider.cpp:123] Check failed: numVecSlots_ == sample.vector_slots_size() (1 vs. 0) 
*** Check failure stack trace: ***
    @           0x79be98  google::LogMessage::Fail()
    @           0x79bdf0  google::LogMessage::SendToLog()
    @           0x79b885  google::LogMessage::Flush()
    @           0x79e646  google::LogMessageFatal::~LogMessageFatal()
    @           0x62fd82  paddle::ProtoDataProvider::checkSample()
    @           0x632737  paddle::ProtoDataProvider::loadDataFile()
    @           0x632db9  paddle::ProtoDataProvider::loadData()
    @           0x6330a2  paddle::ProtoDataProvider::loadData()
    @           0x6334ec  paddle::ProtoDataProvider::ProtoDataProvider()
    @           0x581277  _ZNSt17_Function_handlerIFPN6paddle12DataProviderENS0_10DataConfigEbEZNS0_14ClassRegistrarIS1_IS3_bEE13registerClassINS0_17ProtoDataProviderEEEvRKSsEUlS3_bE_E9_M_invokeERKSt9_Any_dataS3_b
    @           0x581f8b  paddle::ClassRegistrar<>::createByType()
    @           0x57b003  paddle::DataProvider::create()
    @           0x6816a3  paddle::Trainer::init()
    @           0x52a7fe  main
    @     0x7f15b7a1cefd  __libc_start_main
    @           0x5267b5  (unknown)

**Comment by lcy-seso** [comment]
1. 您好，您使用的输入数据格式，以及配置都非常古老，PaddlePaddle 目前已经不再维护这种使用方式，出现错误也不便于排除，建议切换到最新版本。
2. 这个错误的意思是 proto 数据中定义了一个输入slot，但是给进来的某一条样本却是空的，是因为错误的输入数据引起的。
